src/main/scala/org/renci/robocord/json/CORDJsonReader.scala

```diff
@@ -67,31 +67,31 @@ case class ArticleRefEntry(
 )
 
 case class ArticleBibEntry(
-  ref_id: String,
+  ref_id: Option[String],
   title: String,
   authors: Seq[Author],
   year: Option[Int],
   venue: String,
   volume: String,
   issn: String,
-  pages: String,
+  pages: Option[String],
   other_ids: Map[String, Seq[String]]
 )
 
 case class Article(
   paper_id: String,
   metadata: ArticleMetadata,
-  `abstract`: Seq[ArticleAbstract],
+  `abstract`: Option[Seq[ArticleAbstract]],
   body_text: Seq[ArticleBodyText],
   bib_entries: Map[String, ArticleBibEntry],
   ref_entries: Map[String, ArticleRefEntry],
   back_matter: Seq[ArticleBodyText]
 )
 
 class CORDArticleWrapper(article: Article) {
-  val sha1 = article.paper_id
+  val id = article.paper_id
   val titleText = article.metadata.title
-  val abstractText = article.`abstract`.map(_.text).mkString("\n")
+  val abstractText = article.`abstract`.getOrElse(Seq()).map(_.text).mkString("\n")
```

Comment:
I usually use `Seq.empty`, or actually for a concrete empty `Seq` I just use `Nil`.

Reply:
Fixed in 29c65995cd781.